contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -1934,6 +1935,37 @@ class tsqlBuilder : public tsqlCommonMutator
 			/* Re-write session_user to sys.session_user(). */
 			if (bctx->bif_no_brackets && bctx->SESSION_USER())
 				rewritten_query_fragment.emplace(std::make_pair(bctx->bif_no_brackets->getStartIndex(), std::make_pair(::getFullText(bctx->SESSION_USER()), "sys.session_user()")));
+
+			if(bctx->bif_cast_parse())
+			{
+				auto cast_parse = bctx->bif_cast_parse();
+				/* Re-write the expression for casting to time datatype. */
+				if(cast_parse->data_type() && pg_strncasecmp(::getFullText(cast_parse->data_type()).c_str(), "TIME", 4) == 0)
+				{
+					std::string cast_expression(pstrdup(::getFullText(cast_parse->expression()).c_str()));
+
+					/* 
+					 * Remove spaces between before or after {/:.} in the given expression.
```

Comment:
what about handling spaces in pltsql_time_in as well?

Reply:
We can handle this case in `pltsql_time_in`, will move it there.

Reply:
Is it really needed to be moved in `pltsql_time_in`? The functions which were used here are not usable in c, might not get expected result in c.